### PR TITLE
[Testing] Migrate Unit Tests to ARC

### PR DESCRIPTION
**Pull request by Adlai-Holler**
I'm about to start testing the issues we've been having with deallocation, and in order to mimic real conditions in an app I want it to be ARC. So I figured let's get this chore out the way.

Ready for review @maicki @garrettmoon @appleguy @levi

### CODE REVIEW

File: AsyncDisplayKitTests/ASDisplayNodeAppearanceTests.m [hunk 1]
```diff
@@ -104,17 +96,9 @@ - (void)tearDown
...
   [super tearDown];
 
-  for(id cleanupBlock in _swizzleCleanupBlocks) {
-    void (^cleanupBlockCasted)(void) = cleanupBlock;
-    cleanupBlockCasted();
+  for(dispatch_block_t cleanupBlock in _swizzleCleanupBlocks) {
+    cleanupBlock();
   }
-  [_swizzleCleanupBlocks release];
-  _swizzleCleanupBlocks = nil;
```
Comment (garrettmoon) [inline AsyncDisplayKitTests/ASDisplayNodeAppearanceTests.m]: Should we no longer nil?

File: AsyncDisplayKitTests/ASDisplayNodeTests.m [hunk 2]
```diff
@@ -147,14 +141,8 @@ - (void)dealloc
 {
   if (_willDeallocBlock) {
     _willDeallocBlock(self);
-    [_willDeallocBlock release];
     _willDeallocBlock = nil;
```
Comment (garrettmoon) [inline AsyncDisplayKitTests/ASDisplayNodeTests.m]: any reason to nil out _willDeallocBlock?

File: AsyncDisplayKitTests/ASDisplayNodeTests.m [hunk 3]
```diff
@@ -1022,88 +989,53 @@ - (CGPoint)checkConvertPoint:(CGPoint)point toNode:(ASDisplayNode *)toNode selfN
...
-  dispatch_async(queue, ^{
+  dispatch_group_t g = dispatch_group_create();
+  dispatch_group_async(g, queue, ^{
     block();
     blockExecuted = YES;
-    dispatch_semaphore_signal(sema);
   });
-  dispatch_semaphore_wait(sema, DISPATCH_TIME_FOREVER);
-  dispatch_release(sema);
+  dispatch_group_wait(g, DISPATCH_TIME_FOREVER);
```
Comment (garrettmoon) [inline AsyncDisplayKitTests/ASDisplayNodeTests.m]: semaphore's really did kill your parents, huh?

File: AsyncDisplayKitTests/ASDisplayNodeTests.m [hunk 4]
```diff
@@ -1022,88 +989,53 @@ - (CGPoint)checkConvertPoint:(CGPoint)point toNode:(ASDisplayNode *)toNode selfN
...
   XCTAssertTrue(blockExecuted, @"Block did not finish executing. Timeout or exception?");
 }
 
 - (void)testReferenceCounting
```
Comment (garrettmoon) [inline AsyncDisplayKitTests/ASDisplayNodeTests.m]: The existence of this test has horrifying implications.

File: AsyncDisplayKitTests/ASDisplayNodeTests.m [hunk 5]
```diff
@@ -1876,21 +1802,21 @@ - (void)testInitWithLayerClass
   XCTAssertEqual(0.5f, transformNode.alpha, @"Alpha not working");
 }
 
-static bool stringContainsPointer(NSString *description, const void *p) {
-  return [description rangeOfString:[NSString stringWithFormat:@"%p", p]].location != NSNotFound;
+static bool stringContainsPointer(NSString *description, id p) {
+  return [description containsString:[NSString stringWithFormat:@"%p", p]];
```
Comment (garrettmoon) [inline AsyncDisplayKitTests/ASDisplayNodeTests.m]: containsString was introduced in iOS 8 as hard as that is to believe.

File: AsyncDisplayKitTests/ASDisplayNodeTests.m (hunk 3 above)
Comment (Adlai-Holler) [inline AsyncDisplayKitTests/ASDisplayNodeTests.m]: I won't rest until I make those bastards pay

File: AsyncDisplayKitTests/ASDisplayNodeTests.m (hunk 4 above)
Comment (Adlai-Holler) [inline AsyncDisplayKitTests/ASDisplayNodeTests.m]: It's true

**Comment by Adlai-Holler** [comment]
Thanks @garrettmoon, feedback addressed!

**Comment by garrettmoon** [comment]
LGTM!

**Pull request merged**